File: reqquest-backend/reqquest-backend/Models/Collection.cs
namespace reqquest_backend.Models;

public class Collection
{
    public int Id { get; set; }
    public int? ProjectId { get; set; }
    public Project Project { get; set; }
    public string Name { get; set; }
    public List<Request> Requests { get; set; } = new List<Request>();
    public string Description { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}

File: reqquest-backend/entities/collection.go
package entities

import "time"

type Collection struct {
	ID          int       `json:"id"`
	Name        string    `json:"name"`
	Description string    `json:"description"`
	ProjectId   int       `json:"projectId"` //Foreign Key Reference
	Requests    []Request `json:"requests,omitempty"`
	CreatedAt   time.Time `json:"createdAt"`
}
